### PR TITLE
LPS-143100 Multiple Files Upload unnecessarily renames files

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
- LPS-143100 Create two new methods to return the unique fileName and the unique title ignoring the existence of the title/filename for the same return
- LPS-143100 buildService
- LPS-143100 apply the new methods to get the title and the filename when you use the multiupload or the drag and drop to upload files
- LPS-143100 baseline
- LPS-143100 add test to meet the new behavior and rename old test

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portlet/documentlibrary/util/DLImpl.java [hunk 1]
```diff
@@ -873,6 +873,42 @@ public String getUniqueFileName(
...
+
+		while (_existsFileEntryByFileName(
+					groupId, extension, folderId, uniqueFileTitle)) {
+
+			uniqueFileTitle = FileUtil.appendParentheticalSuffix(
+				FileUtil.stripExtension(fileName), String.valueOf(i++));
+		}
+
+		return getTitleWithExtension(uniqueFileTitle, extension);
+	}
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portlet/documentlibrary/util/DLImpl.java:894]: Why is the impl. of these methods different than the already existing one? Choose one option (for+break or while loop) and use that in all methods.

Also, the two getUniqueFile* impls are almos identical -- the only difference is that in the original one, an additional condition is checked. Can you merge both implementations? Something like:
```java

	@Override
	public String getUniqueTitle(
		long groupId, long folderId, String title, boolean ignoreDuplicateTitle) {

		String uniqueFileTitle = title;

		int i = 1;

		while ((ignoreDuplicateTitle ||
				!_existsFileEntryByTitle(groupId, folderId, uniqueFileTitle) &&
			  _existsFileEntryByTitle(groupId, folderId, uniqueFileTitle)) {
			
			uniqueFileTitle = FileUtil.appendParentheticalSuffix(
				title, String.valueOf(i));
			i++;
		}

		return uniqueFileTitle;
	}
```
Make the old method use the new, more general, one (I don't like boolean args either, but it is an established pattern).

Also, if you choose to use a while loop, don't use inline post/pre increments. Put them in their own line (as I did in the previous example).

File: portal-impl/src/com/liferay/portlet/documentlibrary/util/DLImpl.java [hunk 2]
```diff
@@ -873,6 +873,42 @@ public String getUniqueFileName(
...
+
+	@Override
+	public String getUniqueTitleIgnoringFileNameExistence(
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portlet/documentlibrary/util/DLImpl.java:897]: Can we rename this to `getUniqueTitle`?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: portal-impl/src/com/liferay/portlet/documentlibrary/util/DLImpl.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply portal-impl/src/com/liferay/portlet/documentlibrary/util/DLImpl.java:894]: thanks!

File: portal-impl/src/com/liferay/portlet/documentlibrary/util/DLImpl.java (hunk 2 above)
Comment (AliciaGarciaGarcia) [reply portal-impl/src/com/liferay/portlet/documentlibrary/util/DLImpl.java:897]: doing it :)

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
rebase

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by liuxingye** [comment]
ci:test:relevant

**Comment by liuxingye** [comment]
This POSHI LocalFile.OAuth2#GrantedAuthorizationsCanBeRevoked failure is not related to the changes in this PR, also see it on [head](https://testray.liferay.com/home/-/testray/case_results/1594823750) run of master.

**Comment by liuxingye** [comment]
ci:close

**Pull request closed without merging**